keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -0,0 +1,244 @@
+            maxval=tf.maximum(0.0, 1.0 - new_height),
+            dtype=tf.float32,
+        )
+
+        width_offset = self._random_generator.random_uniform(
+            (),
+            minval=tf.minimum(0.0, 1.0 - new_width),
+            maxval=tf.maximum(0.0, 1.0 - new_width),
+            dtype=tf.float32,
+        )
+
+        y1 = height_offset
+        y2 = height_offset + new_height
+        x1 = width_offset
+        x2 = width_offset + new_width
+
+        return [[y1, x1, y2, x2]]
+
+    def call(self, inputs, training=True):
+
+        if training:
+            return super().call(inputs, training)
+        else:
+            inputs = self._ensure_inputs_are_compute_dtype(inputs)
+            inputs, is_dict, use_targets = self._format_inputs(inputs)
+            output = inputs
+            # self._resize() returns valid results for both batched and
+            # unbatched
+            output["images"] = self._resize(inputs["images"])
+            return self._format_output(output, is_dict, use_targets)
+
+    def augment_image(self, image, transformation, **kwargs):
+        image = tf.expand_dims(image, axis=0)
+        boxes = transformation
+
+        # See bit.ly/tf_crop_resize for more details
+        augmented_image = tf.image.crop_and_resize(
+            image,  # image shape: [B, H, W, C]
+            boxes,  # boxes: (1, 4) in this case; represents area
+            # to be cropped from the original image
+            [0],  # box_indices: maps boxes to images along batch axis
+            # [0] since there is only one image
+            self.target_size,  # output size
+        )
+
+        return tf.squeeze(augmented_image, axis=0)
+
+    def _resize(self, image):
+        outputs = tf.keras.preprocessing.image.smart_resize(image, self.target_size)
+        # smart_resize will always output float32, so we need to re-cast.
+        return tf.cast(outputs, self.compute_dtype)
+
+    def _check_class_arguments(
+        self, target_size, crop_area_factor, aspect_ratio_factor
+    ):
+        if not isinstance(target_size, tuple):
+            raise ValueError(
+                "`target_size` must be tuple of two integers."
+                f"Received target_size={target_size}"
+            )
+
+        if len(target_size) != 2:
+            raise ValueError(
+                "`target_size` must be tuple of two integers."
+                f"Received target_size={target_size}"
+            )
+
+        if not isinstance(target_size[0], int):
+            raise ValueError(
+                "`target_size` must be tuple of two integers."
+                f"Received target_size={target_size}"
+            )
+
+        if not isinstance(target_size[1], int):
+            raise ValueError(
+                "`target_size` must be tuple of two integers."
+                f"Received target_size={target_size}"
+            )
+
+        if isinstance(aspect_ratio_factor, core.NormalFactorSampler):
```

Comment:
Using a NormalFactorSampler here is fine - you can remove these checks.  

Please add checks for `aspect_ratio_factor` and `crop_area_factor` being single integers.  Single values are not valid configurations.

check that it is either:
- a factor sampler
- a tuple of two values

Reply:
Oh!  I missed that, let me go review your get_config